frontend/src/components/FacilityBooking/BlockStyles.ts

```diff
@@ -0,0 +1,106 @@
+import styled, { css } from 'styled-components'
+import { TimeBlockType } from '../../store/facilityBooking/types'
+
+export const BLOCK_HEIGHT = 90
+export const TOP_DISTANCE = 10
+export const BLOCK_GAP = 5
+
+export const DailyContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: ${`${TOP_DISTANCE}px`} 5px;
+  gap: ${`${BLOCK_GAP}px`};
+`
+
+const blockStyles = css`
+  position: relative;
+  border-radius: 10px;
+  padding: 10px 15px;
+  color: #000;
+  display: flex;
+  white-space: nowrap;
+  height: ${`${BLOCK_HEIGHT}px`};
+  gap: 10px;
+`
+
+// Styles for Booking Block (Occupied, Available, Selected and Not Available)
+export const StyledBookingBlock = styled.div<{
+  type: TimeBlockType
+  blockId: number
+  hasOverlay?: boolean
+}>`
+  ${(props) => (props.type === TimeBlockType.AVAILABLE || props.type === TimeBlockType.SELECTED) && 'cursor: pointer;'}
+  ${blockStyles}
+  background: ${(props) =>
+    props.type === TimeBlockType.OCCUPIED
+      ? palette.occupied
+      : props.type === TimeBlockType.AVAILABLE
+      ? palette.available
+      : props.type === TimeBlockType.SELECTED
+      ? palette.selected
+      : props.blockId % 2 == 0
+      ? palette.unavailable1
+      : palette.unavailable2};
+
+  ${(props) =>
+    props.hasOverlay &&
+    css`
+      &::after {
+        content: '';
+        position: absolute;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        background: rgba(255, 255, 255, 0.7);
+      }
+    `}
+`
+
+// Styling for View Only Blocks (Occupied, Not Occupied)
+export const StyledViewBooking = styled.div<{ isOccupied: boolean; blockId: number }>`
+  ${(props) => props.isOccupied && 'cursor: pointer;'}
+  ${blockStyles}
+  background: ${(props) =>
+    // occupied is green (same colour as available)
+    props.isOccupied ? palette.available : props.blockId % 2 == 0 ? palette.unavailable1 : palette.unavailable2};
```

Comment:
nitpick point, technically enforced by typescript but triple equals is still preferred

Reply:
o good catch!